en/middleware/micrortps.md

```diff
@@ -30,58 +32,84 @@ The main elements of the architecture are the client and agent processes shown i
+
+
+## Architectural overview for a ROS2/ROS application pipeline
+
+<!--![basic example flow](../../assets/middleware/micrortps/architecture_ros.png)-->
+
+ROS2 has been developed on top of the DDS/RTPS, which is what composes its middleware. This same middleware serves as the end-to-end architecture for plugging different applications that rely on distributed discovery, serialization and QoS control over the transportation layer.
+
+So one is able to publish and subscribe to uORB data from PX4 using ROS2 (and ROS1 if using the [`ros1_bridge`](https://github.com/ros2/ros1_bridge), it just needs to create it's own listener our advertiser nodes in the respective frameworks. The only consideration to take into account is that the msg types, headers and source files being used on both client and agent side (and consequently, on the ROS side) need to be the same and generated from the same IDL (Interface Description Language).
 
 
 ## Code generation
 
-All the code needed to create, build and use the bridge is automatically generated when the PX4 Firmware is compiled. 
+### Code generation for ROS-independent applications
+
+All the code needed to create, build and use the bridge is automatically generated when the PX4 Firmware is compiled.
 
 The *Client* application is also compiled and built into the firmware as part of the normal build process. The *Agent* must be separately/manually compiled for the target computer.
 
 > **Note** [Fast RTPS must be installed](../setup/fast-rtps-installation.md) in order to generate the required code!
 
 <span></span>
-> **Tip** The bridge code can also be [manually generated](micrortps_manual_code_generation.md). Most users will not need to do so, but the linked topic provides a more detailed overview of the build process and can be useful for troubleshooting.
+> **Tip** The bridge code can also be [manually generated](micrortps_manual_code_generation.md). Most users will not need to do
+so, but the linked topic provides a more detailed overview of the build process and can be useful for troubleshooting.
 
+### Code generation for ROS2/ROS applications - `px4_ros_com`
 
-## Supported uORB messages
-
-The generated bridge code will enable a specified subset of uORB topics to be published/subscribed via RTPS.
+The [`px4_ros_com`](https://github.com/PX4/px4_ros_com) package allows the developer to generate all the required ROS2/ROS message headers and sources, but also the agent application which will bridge the client with the RTPS middleware on the ROS2 side. The package is basically composed by two separate branches, where the `master` branch generates all the required ROS2 messages and IDL files, besides some listener and advertiser example nodes, in order to bridge PX4 with ROS2 nodes, and a `ros1` branch, which similarly to the `master` branch, generates the ROS(1) message headers and source files, besides the ROS nodes examples, that allow, by using the `ros1_bridge`, share data between PX4 and ROS.
```

Comment:
I get an HTTP404 on this link https://github.com/PX4/px4_ros_com

Reply:
Probably 

> and sources, but also the 

TO

> and sources, AND also the

Reply:
I will take you suggestion. Much cleaner! Thanks!

> when you say "besides some listener" do you mean "this also has some example nodes"?

I mean the node code that you find on the package are just examples of how people can write the nodes to interact with PX4. So they are just "interesting", not required.